### DISCUSSION TITLE
Reply Interactive Messages

### DISCUSSION THREAD

**Discussion by onktak** [netflie/whatsapp-cloud-api#68] [General]
Hello, 

Thank you for such a great library. It makes integrating with whatsapp cloud api very easy.  What I have realized is missing is the implementation of reply interactive messages. See snippet from the whatsapp docs. I have also realized the header is only handling text but on the header an image or document can be linked. 

`
curl -X  POST \
 'https://graph.facebook.com/v15.0/FROM_PHONE_NUMBER_ID/messages' \
 -H 'Authorization: Bearer ACCESS_TOKEN' \
 -H 'Content-Type: application/json' \
 -d '{
  "messaging_product": "whatsapp",
  "recipient_type": "individual",
  "to": "PHONE_NUMBER",
  "type": "interactive",
  "interactive": {
    "type": "button",
    "body": {
      "text": "BUTTON_TEXT"
    },
    "action": {
      "buttons": [
        {
          "type": "reply",
          "reply": {
            "id": "UNIQUE_BUTTON_ID_1",
            "title": "BUTTON_TITLE_1"
          }
        },
        {
          "type": "reply",
          "reply": {
            "id": "UNIQUE_BUTTON_ID_2",
            "title": "BUTTON_TITLE_2"
          }
        }
      ]
    }
  }
}'
`
It can be handled using templates but its not practical if you have a lot of interactive messages.

**Comment by aalbarca** [comment]
Thanks for your suggestions. 

Right now I'm working on the new 2.x version that will support sending and downloading multimedia files and also webhooks. 

For this first release of the new version it is not programmed to improve the interactive messages but I leave it noted for future releases.

And if you dare to do it, open a branch and improve the code. The community and I will be forever grateful.

Thanks!
